demos/demo_2d_tracer.py

```diff
@@ -0,0 +1,123 @@
+# 2D tracer transport
+# ===================
+#
+# This demo shows how the Firedrake DG advection equation
+# `demo <https://firedrakeproject.org/demos/DG_advection.py.html>`__
+# can be implemented in Thetis.
+#
+# The test case is the classic cosine-bell--cone--slotted-cylinder
+# advection test case of :cite:`LeVeque:1996`. The domain is the unit
+# square :math:`\Omega=[0,1]^2` and the velocity corresponds to the
+# solid body rotation :math:`\vec{u} = (0.5 - y, x - 0.5)`.
+# 
+# As usual, we start by importing Thetis. ::
+
+from thetis import *
+
+# Define a 40-by-40 mesh of squares. ::
+
+mesh2d = UnitSquareMesh(40, 40, quadrilateral=True)
+
+# We will solve a pure advection problem in non-conservative form,
+# with no hydrodynamics. Therefore, bathymetry is not actually
+# important. We set an arbitrary postive value, as this is required
+# by Thetis to construct the solver object. ::
+
+P1_2d = FunctionSpace(mesh2d, "CG", 1)
+bathymetry2d = Function(P1_2d)
+bathymetry2d.assign(1.0)
+solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry2d)
+
+# To activate the tracer functionality, we set the option 
+# ``solve_tracer = True``. As mentioned above, we are only solving
+# the tracer equation, which can be specified by setting
+# ``tracer_only = True``.
+
+options = solver_obj.options
+options.solve_tracer = True
+options.tracer_only = True
+options.fields_to_export = ['tracer_2d']
+
+# We will run for time :math:`2\pi` -- a full rotation -- using a
+# strong stability preserving third order Runge-Kutta method (SSPRK33).
+# We take 600 steps, giving a timestep close to the CFL limit. For
+# consistency with the Firedrake demo, Thetis' automatic timestep
+# computation functionality is switched off and the same solver
+# parameters are specified. ::
+
+options.timestepper_type = 'SSPRK33'
+options.timestep = pi/300.0
+options.simulation_end_time = 2*pi
+options.simulation_export_time = pi/15.0
+options.timestepper_options.use_automatic_timestep = False
+options.timestepper_options.solver_parameters_tracer = {
+    'ksp_type': 'preonly',
+    'pc_type': 'bjacobi',
+    'sub_pc_type': 'ilu',
+}
```

Comment:
Why do we need to specify the solver options here? Do the default ones not work? Or consistency with Firedrake example? For a demo it might be good if we could omit such details...

Reply:
Good point, have removed them.